src/app/examples/todos/todos.reducer.spec.ts

```diff
@@ -0,0 +1,60 @@
+import * as fromTodo from './todos.reducer';
+
+describe('TodoReducer', () => {
+  describe('undefined action', () => {
+    it('should return the default state', () => {
+      const { initialState } = fromTodo;
+      const action = {} as any;
+      const state = fromTodo.todosReducer(undefined, action);
+
+      expect(state).toBe(initialState);
+    });
+  });
+
+  describe('ADD action', () => {
+    it('should add a todo', () => {
+      const { initialState } = fromTodo;
+      const action = new fromTodo.ActionTodosAdd({ name: 'Mercuccio' });
+      const state = fromTodo.todosReducer(initialState, action);
+
+      expect(state.items.length).toEqual(initialState.items.length + 1);
+    });
+  });
+
+  describe('TOGGLE action', () => {
+    it('should toggle selected todo', () => {
+      const { initialState } = fromTodo;
+      const isDone = initialState.items[0].done;
+      const action = new fromTodo.ActionTodosToggle({
+        id: initialState.items[0].id
+      });
+      const state = fromTodo.todosReducer(initialState, action);
+
+      expect(state.items[0].done).not.toEqual(isDone);
+    });
+  });
+
+  describe('REMOVE action', () => {
+    it('should remove done todos', () => {
+      const { initialState } = fromTodo;
+      const action = new fromTodo.ActionTodosRemoveDone();
+      const state = fromTodo.todosReducer(initialState, action);
+      expect(state.items.length).toBe(2);
```

Comment:
Here again we should create test specific initial state with one done todo. Then we emit action to check if there are now 0 items.